cmds/deploy.go

```diff
@@ -256,12 +257,25 @@ func deploy(f *cmdutil.Factory, d DefaultFabric8Deployment) {
 	typeOfMaster := util.TypeOfMaster(c)
 
 	// extract the ip address from the URL
-	ip := strings.Split(cfg.Host, ":")[1]
-	ip = strings.Replace(ip, "/", "", 2)
+	u, err := url.Parse(cfg.Host)
+	if err != nil {
+		util.Fatalf("%s", err)
+	}
+
+	ip, _, err := net.SplitHostPort(u.Host)
```

Comment:
Ah yeah thanks, I'd missed that, when you say check the type of the error, is this good enough:
```
if err != nil && !strings.Contains(err.Error(),"missing port in address") {
   util.Fatalf("%s", err)
}
```